exchanges/gateio/gateio_websocket.go

```diff
@@ -109,14 +115,20 @@ func (g *Gateio) generateWsSignature(secret, event, channel string, t int64) (st
 // WsHandleSpotData handles spot data
 func (g *Gateio) WsHandleSpotData(_ context.Context, respRaw []byte) error {
 	var push WsResponse
-	err := json.Unmarshal(respRaw, &push)
-	if err != nil {
+	if err := json.Unmarshal(respRaw, &push); err != nil {
 		return err
 	}
 
+	if push.RequestID != "" {
+		if !g.Websocket.Match.IncomingWithData(push.RequestID, respRaw) {
+			return fmt.Errorf("%w for requestID %v", stream.ErrNoMessageListener, push.RequestID)
```

Comment:
I have a suggestion we could standardise this at the stream package level and return an error, just wondering if you want this option instead that way if the requirement is there we just return the error and it will give back all the details without having to worry about it wrapper side, let me know what you think:

```go 
func (m *Match) IncomingWithDataMust(signature any, data []byte) error {
	if m.IncomingWithData(signature, data) {
		return nil
	}
	return fmt.Errorf("'%v' %w with data %v", signature, errSignatureNotMatched, string(data))
}
```

Reply:
Agreed.
Note: I think that `ErrNoMessageListener` is widespread and should replace `errSignatureNotMatched` in your example